lib/host_runner.rb

```diff
@@ -35,10 +47,60 @@ def start
       raise LockError.new("Host '#{@host_name}' already locked")
     end
 
+    connect
```

Comment:
`connect` also checks that the SSH connection works, and that is required in any case. We can extract the snapper check into a separate method and only run that if cleanup is activated, though.